File: pet_backend/src/main/java/com/pat/product/entity/Product.java
package com.pat.product.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import com.pat.common.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("product")
public class Product extends BaseEntity {

    private Long storeId;

    private String productName;

    private Integer productType;

    private String category;

    private String productDesc;

    private BigDecimal price;

    private Integer stock;

    private String mainImage;

    private String images;

    private Integer status;

    private Long videoId;

    @TableLogic(value = "0", delval = "1")
    @TableField("deleted")
    private Integer deleted;
}

File: pet_admin/src/types/product.ts
export interface Product {
  id: string;
  productName?: string;
  productType?: number;
  category?: string;
  productDesc?: string;
  detail?: string;
  price: number;
  stock: number;
  status?: string | number;
  statusCode?: number;
  mainImage?: string;
  image?: string;
  images?: string;
  storeId: string;
  name?: string;  // alias for productName
  videoId?: number;
  offlineReason?: string;
  offlineUserId?: string;
  offlineTime?: string;
  platformRestricted?: boolean;
  store?: {
    id: string;
    storeName: string;
    status?: number;
    statusText?: string;
  };
  createTime: string;
}
